tests/unit/providers/test_gemini_provider.py

```
@@ -1523,6 +1527,159 @@ def test_convert_response_no_thought_signature() -> None:
+                    "extra_content": {"google": {"thought_signature": signature}},
+                }
+            ],
+        }
+    ]
+
+    with pytest.raises(InvalidRequestError, match="thought_signature"):
+        _convert_messages(messages)
+
+
+@pytest.mark.parametrize("signature", ["dGVzdA==", "dGVzdA", "-_8-P76_", b"test-signature-bytes"])
+def test_convert_messages_accepts_every_signature_spelling_the_sdk_decodes(signature: Any) -> None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `signature`

(ANN401)